{%extends "componenttib/base.html"%}


<!-- head -->
{%block head_link%}
    {{block.super}}
{%endblock%}
{%block head_script%}
	{{block.super}}
	<script type="text/javascript">
	//模型文件选择还是提交
	function modelUseTypeClick(triggerObj){
		if (triggerObj.value=="select"){//backup if (effectedObj.style.display=="none"){
			//select
			document.querySelector("#selectModelDiv").style.display="block";
			document.querySelector("#uploadModelP").style.display="none";
			//effectedObj.style.display="inline";
			//backup triggerObj.innerHTML="Hide";
			//backup triggerObj.value="Hide";
		}else{
			//upload
			document.querySelector("#selectModelDiv").style.display="none";
			document.querySelector("#uploadModelP").style.display="block";
			//backup triggerObj.innerHTML="Show";
			//backup triggerObj.value="Show";
			//svg.parentNode.removeChild(svg);//backup
		}
	}
	</script>
{%endblock%}

{%block head_title%}{{queryName}}{%endblock%}

<!-- CONTENT -->
{%block content%}


<div style="margin:0px 0px 50px 0px;width:100%;">
	<div style="margin:50px 0px 10px 0px;width:100%;padding-left:200px;"><h3>Online tool for designing <b>g</b>RNA for <b>B</b>ase editor-mediated <b>I</b>nactivation of <b>G</b>enes (gBIG)</h3>
		<p>gBIG designs guide RNAs for base editing-mediated inactivation of given genes in a given sequenced genome. </p>
	<!--backup <div>
		<p>Component sequence <button type="button" class="button blue" onclick="triggerShow5Hide('#show8Word',this)" value="Hide">Hide</button></p>
		<div id="show8Word"></div>-->
	</div>
		<form enctype="multipart/form-data" action="{% url 'sgrna' 'gBIG' %}" method="post">
			{% csrf_token %}    
	<div style="margin:50px 0px 50px 0px;width:50%;padding-left:200px;padding-right:100px;float:left;">

		<fieldset>
			<legend>PAM Type</legend>
				<div class="form-group">
					<div style="margin-left: 20px; margin-right: 20px;">
						<p><b>CRISPR/Cas variant </b></p>
					</div>
					<div style="margin-left: 30px; margin-right: 20px;">
						<div class="radio">
							<label><input type="radio" name="pam_type" id="sgRNAcas9-NGG" value="sgRNAcas9-NGG" checked /> SpCas9 from Streptococcus pyogenes: 5'-NGG-3'</label>
						</div>
						
						<div class="radio">
							<label><input type="radio" name="pam_type" id="sgRNAcas9-NAG" value="sgRNAcas9-NAG"  /> SpCas9 from Streptococcus pyogenes: 5'-NAG-3'</label>
						</div>   
						
						<div class="radio">
							<label><input type="radio" name="pam_type" id="sgRNAcas9-NGCG" value="sgRNAcas9-NGCG"  /> VRER SpCas9 from Streptococcus pyogenes: 5'-NGCG-3'</label>
						</div>
						
						<div class="radio">
							<label><input type="radio" name="pam_type" id="sgRNAcas9-NGA" value="sgRNAcas9-NGA"  /> VQR SpCas9 from Streptococcus pyogenes: 5'-NGA-3'</label>
						</div>
						
						<div class="radio">
							<label><input type="radio" name="pam_type" id="sgRNAcas9-NGT" value="sgRNAcas9-NGT"  /> xCas9 3.7 (TLIKDIV SpCas9) from Streptococcus pyogenes: 5'-NGT-3'</label>
						</div>
						
						<div class="radio">
							<label><input type="radio" name="pam_type" id="sgRNAcas9-NGN" value="sgRNAcas9-NGN"  /> SpCas9-NG (VRVRFRR SpCas9) from Streptococcus pyogenes: 5'-NG-3'</label>
						</div>
					</div>
				</div>
		</fieldset>
		<fieldset>
			<legend>Base Editor Type</legend>
				<div style="margin-left: 30px; margin-right: 20px;">
					<div style="margin:0px 40px 0px 40px;">
						<div class="radio">
							<label><input type="radio" name="base_editor_type" id="Target_AID1" value="Target_AID1" checked /><b>Target-AID<sup> 1</sup></b></label>
						</div>
						<img src="/componenttib/static/image/Target_AID.png" height="70"/>
					</div>
					<div style="margin:0px 40px 0px 40px;">
						<div class="radio">
							<label><input type="radio" name="base_editor_type" id="BE2" value="BE2"  /><b>BE<sup> 2</sup></b></label>
						</div>
						<img src="/componenttib/static/image/BE.png" height="70"/>
					</div>
					<p><sup>1</sup> Nishida K, Arazoe T, Yachie N, Banno S, Kakimoto M, Tabata M, Mochizuki M, Miyabe A, Araki M, Hara KY, Shimatani Z, Kondo A. 2016. Targeted nucleotide editing using hybrid prokaryotic and vertebrate adaptive immune systems. Science 353:aaf8729.</p>
					<p><sup>2</sup> Komor AC, Kim YB, Packer MS, Zuris JA, Liu DR. 2016. Programmable editing of a target base in genomic DNA without double-stranded DNA cleavage. Nature 533:420–424.</p>
				</div>
		</fieldset>
	</div>
	<div style="margin:50px 0px 50px 0px;width:50%;padding-left:100px;padding-right:200px;float:right;">
        <fieldset>
            <legend>Reference Genome</legend>
            <p>Chose <span title="select or upload reference genome">{{form.modelUseType.errors}}{{form.modelUseType}}</span></p>
            <p id="uploadModelP" style="display:none">Upload reference genome (Fasta format)<span title="reference genome file">{{form.referenceFile8Upload.errors}}{{form.referenceFile8Upload}}</span></p>
            <div id="selectModelDiv" style="display:block">
                <p>Select reference genome <button type="button" class="button blue" onclick="triggerShow5Hide('#modelTabDiv',this)" value="Hide" style="width:70px;">Hide</button></p>
                <div id="modelTabDiv">
                <table class="gridtable">
                    <div style="margin-left: 20px; margin-right: 20px;">
                        <p><b>Genomes</b></p>
                    </div>
                    <div style="margin-left: 30px; margin-right: 20px;">
                        
                    <div class="radio">
                        <label><input type="radio" name="ref_seq" id="ASM584v2" value="ASM584v2" checked /> Escherichia coli str. K-12 substr. MG1655</label>
                    </div>
                    <div class="radio">
                        <label><input type="radio" name="ref_seq" id="ASM1132v1" value="ASM1132v1"  /> Corynebacterium glutamicum ATCC 13032</label>
                    </div>
                    <div class="radio">
                        <label><input type="radio" name="ref_seq" id="ASM904v1" value="ASM904v1"  /> Bacillus subtilis 168</label>
                    </div>
                    <div class="radio">
                        <label><input type="radio" name="ref_seq" id="NC_001136.10" value="NC_001136.10"  /> Saccharomyces cerevisiae S288C</label>
                    </div>
					<div class="radio">
                        <label><input type="radio" name="ref_seq" id="GRCm38.p6" value="GRCm38.p6"  /> Mus musculus (GRCm38.p6)</label>
                    </div>
					<div class="radio">
                        <label><input type="radio" name="ref_seq" id="GRCh38.p12" value="GRCh38.p12"  /> Homo sapiens (GRCh38.p12)</label>
                    </div>
					<div class="radio">
                        <label><input type="radio" name="ref_seq" id="hs_ref_GRCh38.p12_chr1" value="hs_ref_GRCh38.p12_chr1"  /> Homo sapiens chr1(GRCh38.p12)</label>
                    </div>
                    </div>
                </table>
				</div>
			</div>
        </fieldset>
		<fieldset>
			<legend>Upload target gene</legend>
			<p>Fasta format. Sequence of expressed region is required for eukaryotic genes.</p>
			<p>Upload inactivated target gene <span title="target gene file">{{form.geneFile8Upload.errors}}{{form.geneFile8Upload}}</span></p>
			<p>Email address <span title="Email address">{{form.userEmail.errors}}{{form.userEmail}}</span></p>
			<p><button type="submit" class="button blue" value="Design">Design</button></p>
		</fieldset>
		
    </div>
    </form>
</div>

<div style="margin:50px 0px 50px 0px;width:100%;padding-left:200px;padding-right:200px;float:left;">
	<h3>User Guide to gBIG</h3>
	<h4><b>gRNA design process on gBIG</b></h4>
	<img src="/componenttib/static/image/gRNA design process on gBIG.png" height="400"/>
	<h4><b>Inputs</b></h4>
	<p>gBIG supports DNA sequence inputs in the form FASTA files. <b>Space</b> and <b>"_"</b> should be avoided in the target gene name.</p>
	<p>Example for target gene:</p>
	<p>>Cgl0413</p>
	<p>ATGGGTTCTGTCATCAAGAAGCGCCGCAAGCGCATGTCCAAGAAGAAGCACCGCAAGATGCTGCGCCGTACTCGTGTCCAGCGTAGAAAATTGGGCAAGTA...</p>
	<p>Example for reference genome:</p>
	<p>>Corynebacterium_glutamicum_ATCC13032</p>
	<p>GTGAGCCAGAACTCATCTTCTTTGCTCGAAACCTGGCGCCAAGTTGTTGCCGATCTCACAACTTTGAGCCAGCAAGCGGACAGTGGATTCGACCCATTGAC...</p>
	<h4><b>Outputs</b></h4>
	<p>There are 31 columns in the output file <b>"sgRNA_result_editable.csv"</b></p>
	<table>
		<tr>
			<th scope="col">Item</th>
			<th scope="col">Info</th>
		</tr>
		<tr>
			<td>sgRNA_ID</td>
			<td>ID for each sgRNA</td>
		</tr>
		<tr>
			<td>Start</td>
			<td>The start position of sgRNA in the target gene</td>
		</tr>
		<tr>
			<td>End</td>
			<td>The end position of sgRNA in the target gene</td>
		</tr>
		<tr>
			<td>CRISPR_target_sequence (5'-3')</td>
			<td>CRISRP target sequence (5'-3'), containing PAM</td>
		</tr>
		<tr>
			<td>sgRNA_Sequence</td>
			<td>sgRNA sequence (5'-3')</td>
		</tr>
			<td>PAM_Sequence</td>
			<td>PAM sequence</td>
		</tr>
		<tr>
			<td>CRISPR_target_sequence_Length (nt)</td>
			<td>Length of CRISPR target sequence containing PAM (nt)</td>
		</tr>
		<tr>
			<td>GC%</td>
			<td>GC content of CRISPR target sequence</td>
		</tr>
		<tr>
			<td>1st_target_codon</td>
			<td>The first target codon</td>
		</tr>
		<tr>
			<td>Editable_for_forming_an_early_stop_codon</td>
			<td>"1" represents this codon can be edited to form an early stop codon. "0" represents this codon can be edited but no early stop codon will be generated</td>
		</tr>
			<td>Position_of_C_inside_gRNA</td>
			<td>Position of the target C inside the gRNA</td>
		</tr>
		<tr>
			<td>Position_of_C_inside_the_gene (%)</td>
			<td>Position of the target C inside the gene (%)</td>
		</tr>
		<tr>
			<td>2nd_target_codon</td>
			<td>The second target codon</td>
		</tr>
		<tr>
			<td>Editable_for_forming_an_early_stop_codon</td>
			<td>"1" represents this codon can be edited to form an early stop codon. "0" represents this codon can be edited but no early stop codon will be generated</td>
		</tr>
		<tr>
			<td>Position_of_C_inside_gRNA</td>
			<td>Position of the target C inside the gRNA</td>
		</tr>
		<tr>
			<td>Position_of_C_inside_the_gene (%)</td>
			<td>Position of the target C inside the gene (%)</td>
		</tr>
		<tr>
			<td>M0_OT</td>
			<td>On-target site. 0M_OT>1 represents repeat genes or off-target sites with 0 mismatch</td>
		</tr>
		<tr>
			<td>M1_OT</td>
			<td>No. of off-target sites with 1 mismatch</td>
		</tr>
		<tr>
			<td>M2_OT</td>
			<td>No. of off-target sites with 2 mismatches</td>
		</tr>
			<td>M3_OT</td>
			<td>No. of off-target sites with 3 mismatches</td>
		</tr>
		<tr>
			<td>M4_OT</td>
			<td>No. of off-target sites with 4 mismatches</td>
		</tr>
		<tr>
			<td>M5_OT</td>
			<td>No. of off-target sites with 5 mismatches</td>
		</tr>
		<tr>
			<td>Total_No.of_OT</td>
			<td>Total No. of off-target sites</td>
		</tr>
		<tr>
			<td>M0_POT</td>
			<td>Perfect match to the 12-nt seed region of the sgRNA and 0 mismatch in the outer segments</td>
		</tr>
		<tr>
			<td>M1_POT</td>
			<td>Perfect match to the 12-nt seed region of the sgRNA and 1 mismatch in the outer segments</td>
		</tr>
		<tr>
			<td>M2_POT</td>
			<td>Perfect match to the 12-nt seed region of the sgRNA and 2 mismatches in the outer segments</td>
		</tr>
		<tr>
			<td>M3_POT</td>
			<td>Perfect match to the 12-nt seed region of the sgRNA and 3 mismatches in the outer segments</td>
		</tr>
		<tr>
			<td>M4_POT</td>
			<td>Perfect match to the 12-nt seed region of the sgRNA and 4 mismatches in the outer segments</td>
		</tr>
			<td>M5_POT</td>
			<td>Perfect match to the 12-nt seed region of the sgRNA and 5 mismatches in the outer segments</td>
		</tr>
		<tr>
			<td>Total_No.of_POT</td>
			<td>Total number of off-targets sites with perfect match to the 12-nt seed region of the sgRNA</td>
		</tr>
		<tr>
			<td>Risk_evaluation</td>
			<td>Evaluation of off-target risk</td>
		</tr>
	</table>
	<p>
	<b>Notes:</b> OT stands for off-targets, which contain all types of off-target sites; POT stands for potential off-target, only contain off-targets which are
	perfect match to the 12-nt seed region of the guide RNA, while outer segments contain mismatched bases.
	</p>
	<p>Off-target risk of guide RNA:</p>
	<p>Discard > High_risk > moderate_risk> low_risk > repeat_sites_or_bad > Best</p>
	<p><b>Notes:</b> gBIG is free for academic, nonprofit, and personal use.</p>
	<p>If you have any questions, please contact us: yangyi@tib.cas.cn.</p>

	{{httpString | safe}}	
</div>


{%endblock%}